### PR TITLE
Feat : Added instructions for running testnet node in Windows

### PR DESCRIPTION
**Pull request by HariniRajan397**
…miner node) on Windows using Binary.

Fixes blockstack/docs#661

## Description
Added instructions for running the testnet node (Both follower and miner node) on Windows from the binary available. 

1. Motivation for change
Most of the docs are Linux users oriented and hence added instructions for Windows users. 

3. How does this impact application developers
This helps and motivates Windows users to get started with their testnet node. 

4. Link to relevant issues and documentation

https://github.com/blockstack/docs/issues/661 - This is the relevant issue and the refernce is from 
https://github.com/blockstack/docs/issues/661#issuecomment-723516837 as well as from https://discordapp.com/channels/621759717756370964/728030782693834862/774786060738887681.

## Type of Change
- [x] New feature

## Does this introduce a breaking change?
No

- [ ] Link to documentation updates: 

## Testing information

Provide context on how tests should be performed.

1. Is testing required for this change?
 This is tested on Windows and if needed, one more round of testing in a fresh Windows machine can be done. 

## Checklist
@agraebe @whoabuddy @zrixes

### CODE REVIEW

**Comment by HariniRajan397** [comment]
@whoabuddy and @zrixes I have added teh instructions for Windows and created a pull request. Sincere thanks to both of you !

File: src/pages/mining.md [hunk 1]
```diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
 BLOCKSTACK_DEBUG=1 stacks-node krypton
 ```
 
+## Running a miner in Windows
+
```
Comment (agraebe) [inline src/pages/mining.md:130]: are there any requirements that we should point out?

File: src/pages/mining.md [hunk 2]
```diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
...
+### Generate keychain and get testnet tokens in Windows
+
+To setup the miner, first, we need to generate a keychain. With this keychain, we'll get some testnet BTC from a faucet, and then use that BTC to start mining.
+
+To get a keychain, the simplest way is to use the `stacks-cli`.We'll use the `stx make-keychain` command,and pass -t to indicate that we want a testnet keychain.
```
Comment (agraebe) [inline src/pages/mining.md:135]: "use the `stacks-cli`. We'll use the" (spacing after dot). makes me think: could you please run a linter on this file and fix linting issues?

File: src/pages/mining.md [hunk 3]
```diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
...
+    "address": "STTX57EGWW058FZ6WG3WS2YRBQ8HDFGBKEFBNXTF",
+    "btcAddress": "mkRYR7KkPB1wjxNjVz3HByqAvVz8c4B6ND",
+    "index": 0
+  }
+}
+```
+
+Request BTC from faucet:
+
+We need to get some testnet BTC to that address. Grab the btcAddress field, and call the BTC faucet:
```
Comment (agraebe) [inline src/pages/mining.md:161]: btcAddress > `btcAddress`

File: src/pages/mining.md [hunk 4]
```diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
...
+
+```bash
+# replace <btc_address> with `btcAddress` property from your keychain
+curl -XPOST "https://stacks-node-api.blockstack.org/extended/v1/faucets/btc?address=<btc_address>"
```
Comment (agraebe) [inline src/pages/mining.md:165]: I think I remember someone struggling with curl on windows - could we add it as a req and link to a place to install it on windows?

File: src/pages/mining.md [hunk 5]
```diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
...
+```
+
+You'll be sent 0.5 testnet BTC to that address. Don't lose this information - we'll need to use the `privateKey` field later on.
+
+### Create configuration file
+
+Now, we need to configure our node to use this Bitcoin keychain. In the `folder where your binary is extracted`, create a new file called `testnet-miner-conf.toml`.
```
Comment (agraebe) [inline src/pages/mining.md:172]: `folder where your binary is extracted` > folder where your binary is extracted

(we should use `var` for code/var highlights mostly)

File: src/pages/mining.md (hunk 2 above)
Comment (agraebe) [reply src/pages/mining.md:135]: "command,and pass -t" > "command, and pass `-t`" (spacing and code highlight)

File: src/pages/mining.md [hunk 6]
```diff
@@ -126,6 +126,108 @@ In case you are running into issues or would like to see verbose logging, you ca
...
+To setup the miner, first, we need to generate a keychain. With this keychain, we'll get some testnet BTC from a faucet, and then use that BTC to start mining.
+
+To get a keychain, the simplest way is to use the `stacks-cli`.We'll use the `stx make-keychain` command,and pass -t to indicate that we want a testnet keychain.
+
+Generate a keychain:
+
+```bash
+npm install --global @stacks/cli
+stx make_keychain -t > cli_keychain.json
+type cli_keychain.json
```
Comment (agraebe) [inline src/pages/mining.md:142]: so, if I read this correctly, this is the most divergence from the non-windows steps, correct? if it is just that - we should rather make a note in the other tutorial and avoid duplicating 90% of content from the other tutorial. wdyt @HariniRajan397 ?

File: src/pages/stacks-blockchain/running-testnet-node.md [hunk 7]
```diff
@@ -91,6 +91,51 @@ INFO [1588108047.585] [src/chainstate/stacks/index/marf.rs:732] First-ever block
...
+#### Optional Dependencies
+
+- [Python](https://www.python.org/downloads/).
+- [Rust](https://www.rust-lang.org/tools/install).
+
+### Download the Binary and run the follower node
+
+First, download the binary using the below link:
+
+<https://github.com/blockstack/stacks-blockchain/releases/download/v23.0.0.10-krypton/windows-x64.zip>
```
Comment (agraebe) [inline src/pages/stacks-blockchain/running-testnet-node.md:113]: I'd rather do this: "[download the binary](https://github.com/blockstack/stacks-blockchain/releases/download/v23.0.0.10-krypton/windows-x64.zip)"

File: src/pages/stacks-blockchain/running-testnet-node.md [hunk 8]
```diff
@@ -91,6 +91,51 @@ INFO [1588108047.585] [src/chainstate/stacks/index/marf.rs:732] First-ever block
...
+
+Extract the binary .Open the command prompt from the folder where binary is extracted and execute the below command:
```
Comment (agraebe) [inline src/pages/stacks-blockchain/running-testnet-node.md:115]: "binary .Open" > "binary. Open" (spacing)

-> again, would run a linter. I would assume it would fail with the current sources and we'd want to fix these issues. Failing lint tests would break the deployment of the PR

**Review by agraebe — COMMENTED** [review]
thanks for submitting the PR. Windows instructions will be very useful, so this is very impactful. I left some notes.

tl;dr: fix linting issues & explorer if we can put windows-specific (divergent) steps into the existing tutorial (most of the content seems very similar to non-windows instructions)

File: src/pages/mining.md (hunk 3 above)
Comment (HariniRajan397) [reply src/pages/mining.md:161]: Fixed it.

File: src/pages/mining.md (hunk 1 above)
Comment (HariniRajan397) [reply src/pages/mining.md:130]: Updated the requirements.

File: src/pages/mining.md (hunk 4 above)
Comment (HariniRajan397) [reply src/pages/mining.md:165]: Agree. Added the appropriate links to install curl on Windows.

File: src/pages/mining.md (hunk 5 above)
Comment (HariniRajan397) [reply src/pages/mining.md:172]: Fixed it.

File: src/pages/stacks-blockchain/running-testnet-node.md (hunk 7 above)
Comment (HariniRajan397) [reply src/pages/stacks-blockchain/running-testnet-node.md:113]: Agreed. Updated the details along with the note.

File: src/pages/stacks-blockchain/running-testnet-node.md (hunk 8 above)
Comment (HariniRajan397) [reply src/pages/stacks-blockchain/running-testnet-node.md:115]: Fixed the linting issues.

File: src/pages/mining.md (hunk 2 above)
Comment (HariniRajan397) [reply src/pages/mining.md:135]: Fixed the same .

**Review by HariniRajan397 — COMMENTED** [review]
Incorporated the review comments and updated the repo.

**Review by CharlieC3 — COMMENTED** [review]
Just tested these instructions out on my Windows machine and they worked great! I followed the instructions to the letter for both running a follower node and a miner, and doing that helped me find a couple areas where users may need some extra assistance.

1. When installing MS BuildTools, the C++ package is not selected by default. It might be worth adding a small excerpt next to this prereq telling the user they'll need to manually select the C++ build tools option when installing.

1. After first starting a follower node, I received the below Windows Defender warning. We should add an excerpt explaining the user may see a Windows Defender warning when first running the node, and to be sure to select "Allow Access" when/if prompted. This excerpt should be added to both the testnet and miner pages.

**Comment by HariniRajan397** [comment]
Thanks, @CharlieC3 ! Incorporated the review comments and added the instructions with an accompanying snapshot for the C++ Build tools as well as Windows Defender popup. Kindly re-review the latest commit 
[c64b4b6 ](https://github.com/blockstack/docs/pull/891/commits/c64b4b6aaaeb5d8716e43d87201cec86b10d5bae). Thanks.

cc @agraebe

**Review by agraebe — APPROVED** [review]
looks great, thanks @HariniRajan397 !

**Review by CharlieC3 — APPROVED** [review]
I like the additions you made! :shipit:
